packages/cubejs-client-core/index.d.ts

```diff
@@ -680,11 +680,21 @@ declare module '@cubejs-client/core' {
 
   export type TimeDimensionGranularity = 'second' | 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year';
 
-  export type TimeDimension = {
+  export interface TimeDimensionBase {
     dimension: string;
-    dateRange?: string | string[];
     granularity?: TimeDimensionGranularity;
-  };
+  }
+
+  export interface TimeDimensionComparison extends TimeDimensionBase {
+    compareDateRange: Array<string | Array<string>>;
+    dateRange?: never;
+  }
+
+  export interface TimeDimensionRanged extends TimeDimensionBase {
+    dateRange?: string | string[];
```

Comment:
I think the second option should be `[string, string]`

Reply:
Should it? It was `dateRange?: string | string[];` in the original code.

Reply:
It was incorrect then. `dateRange` is either a string or an array of two dates.